sympy/core/symbol.py

```diff
@@ -17,6 +17,8 @@
 import re as _re
 import random
 from itertools import product
+from typing import List, Tuple as tTuple, Any
```

Comment:
I think it is better to start using an alias for List as well since there is a List-class in SymPy. Even better to create a `utilities.typing` with these aliases and import from there.

Reply:
I took a look at https://www.python.org/dev/peps/pep-0585/#id5
It seems better to directly use `list` with `from __future__ import annotation`, as `from typing import List` is deprecated.

Reply:
Yes, let's do this. I don't like these `tTuple` things as they are confusing. This should be fixed everywhere in the codebase, though, so it can be done in a separate PR. I think we can also use the `|` syntax. It just isn't supported if you write it directly in Python  < 3.10 (i.e., without using `from __future__ import annotations), but mypy itself will support it in any Python version. `from __future__ import annotations` also means we can convert string and comment annotations into real annotations.